### PR TITLE
Fix ClassCastException in TextInput

### PR DESCRIPTION
**Pull request by guptasourabh**
ClassCastException happening if getText() is not Spanned 

Fatal Exception: java.lang.ClassCastException: java.lang.String cannot be cast to android.text.Spanned
       at com.facebook.react.views.text.ReactTextView.reactTagForTouch(ReactTextView.java:52)
       at com.facebook.react.uimanager.TouchTargetHelper.getTouchTargetForView(TouchTargetHelper.java:257)
       at com.facebook.react.uimanager.TouchTargetHelper.findTargetTagAndCoordinatesForTouch(TouchTargetHelper.java:101)
       at com.facebook.react.uimanager.JSTouchDispatcher.handleTouchEvent(JSTouchDispatcher.java:76)

Test plan: Manual code inspection, Circle CI.

### CODE REVIEW

**Comment by guptasourabh** [comment]
Signed it

**Comment by guptasourabh** [comment]
@mkonicek for review

**Comment by mkonicek** [comment]
The fix looks good. Any idea why it's not an instance of `Spanned`?

**Comment by guptasourabh** [comment]
@mkonicek We are not able to reproduce it from our side ( we tested on 100 devices ), but we are seeing the crash happening on very random devices in our Fabric dashboard. 
screen shot 2017-03-15 at 11 26 57 am

**Comment by guptasourabh** [comment]
@mkonicek Can you please review it. As this fix will help us to reduce the number if crashes happening in app due to react native

**Comment by guptasourabh** [comment]
@mkonicek Can you please review it. As this fix will help us to reduce the number if crashes happening in app due to react native

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java [hunk 1]
```diff
@@ -79,18 +79,19 @@ public void setText(ReactTextUpdate update) {
...
-    int target = getId();
-
-    int x = (int) touchX;
-    int y = (int) touchY;
-
     Layout layout = getLayout();
-    if (layout == null) {
+    int target = getId();
+    
+    if (layout == null || !(getText() instanceof Spanned)) {
```
Comment (shergin) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java]: Can we do not call `getText` twice here?

**Pull request closed without merging**